irohad/model/converters/json_common.hpp

```diff
@@ -18,35 +18,336 @@
 #ifndef IROHA_JSON_COMMON_HPP
 #define IROHA_JSON_COMMON_HPP
 
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+
+#define RAPIDJSON_HAS_STDSTRING 1
 #include <rapidjson/document.h>
-#include <initializer_list>
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
 #include <nonstd/optional.hpp>
-#include <string>
+
+#include "common/byteutils.hpp"
+#include "model/block.hpp"
+#include "model/common.hpp"
 #include "model/signature.hpp"
 
 namespace iroha {
   namespace model {
     namespace converters {
+      /**
+       * Create map get function for value retrieval by key
+       * @tparam K - map key type
+       * @tparam V - map value type
+       * @param map - map for value retrieval
+       * @return function which takes key, returns value if key exists,
+       * nullopt otherwise
+       */
+      template <typename K, typename V>
+      auto makeOptionalGet(std::unordered_map<K, V> map) {
+        return [&map](auto key) -> nonstd::optional<V> {
+          auto it = map.find(key);
+          if (it != std::end(map)) {
+            return it->second;
+          }
+          return nonstd::nullopt;
+        };
+      }
 
-      bool verifyRequiredMembers(
-          const rapidjson::Document& document,
-          const std::initializer_list<std::string>& members);
+      /**
+       * Return function which invokes class method by pointer to member with
+       * provided arguments
+       * @tparam T - provided class type
+       * @tparam Args - provided arguments types
+       * @param object - class object
+       * @param args - function arguments
+       * @return described function
+       */
+      template <typename T, typename... Args>
+      auto makeMethodInvoke(T &object, Args &&... args) {
+        return [&](auto f) {
+          return (object.*f)(std::forward<Args>(args)...);
+        };
+      }
 
-      rapidjson::Document serializeSignature(const Signature& signature);
+      /**
+       * Convert functor which specifies output type
+       * @tparam V - output type
+       */
+      template <typename V>
+      struct Convert {
+        /**
+         * Convert input type to defined type
+         * @tparam T - input type
+         * @param x - input value
+         * @return optional of output type from input value
+         */
+        template <typename T>
+        auto operator()(T &&x) {
+          return nonstd::optional<V>(x);
+        }
+      };
 
-      nonstd::optional<Signature> deserializeSignature(
-          const rapidjson::Document& document);
+      template <size_t size>
+      struct Convert<blob_t<size>> {
+        template <typename T>
```

Comment:
Maybe it is better to put string here not general T, since you are using hexstringToArray

Reply:
`T&&t` can be any iterable type. Vector, string, array, etc.